README.md

```diff
@@ -1,15 +1,27 @@
 # Recursion Problems
 
 ## Definitions
-Define the following: 
+Define the following:
 
 - Recursion
+Recursion is where a function calls on itself, the solution to the problem may consist of smaller instances of the same function.
+
 - Recursive Case
+The recursive case is an instance where a function will call on itself for the result.
 - Base Case
+The base case is the lowest level case that the recursive case works towards. If the base case is never reached, the recursion will become an infinite recursion.
+
 - Activation Chain/Stack
+The Activation Chain or stack is the series of functions that were required to arrive at the result.
+
 - Activation Record/Call
+The activation record is the memory that contains all the information to keep track of the functions being run, these consist of parameters, variables, return values, etc.
+
 - Infinite Recursion/Stack Overflow/Stack too deep
+This means that the recursion is calling itself infinitely and a solution can't be reached (or that no solution is possible).
+
 - Tail Recursion
+I think that tail recursion means that rather than performing all your recursion calls first before returning back to calculate the results, you first perform all your calculations then move to the next recursive step. In this way, some efficiency is gained as you don't need to go through the activation chain all over again to perform the recursion.
```

Comment:
I didn't go over this in class, but this is mostly right. I may cover in the future, but it's not terribly important at this moment.